internal/controller/management/projects/projects.go

```diff
@@ -392,6 +400,74 @@ func (r *reconciler) ensureAPIAdminPermissions(
 	return nil
 }
 
+func (r *reconciler) ensureControllerPermissions(
+	ctx context.Context,
+	project *kargoapi.Project,
+) error {
+	logger := logging.LoggerFromContext(ctx).WithValues(
+		"project", project.Name,
+		"namespace", project.Name,
+	)
+
+	// Get all ServiceAccounts labeled as controller ServiceAccounts
+	controllerSAs := &corev1.ServiceAccountList{}
+	if err := r.client.List(ctx, controllerSAs, client.InNamespace(r.cfg.KargoNamespace), client.MatchingLabels{"app.kubernetes.io/component": "controller"}); err != nil {
+		return fmt.Errorf("error listing controller ServiceAccounts: %w", err)
+	}
+
+	// Create/update a RoleBinding for each ServiceAccount
+	for _, sa := range controllerSAs.Items {
+		roleBindingName := fmt.Sprintf("kargo-controller-secrets-readonly-%s", sa.Name)
```

Comment:
I would suggest a slightly different naming convention here:

```suggestion
		roleBindingName := fmt.Sprintf("%-readonly-secrets", sa.Name)
```